Sources/XcodeGenKit/SourceGenerator.swift

```diff
@@ -259,26 +261,73 @@ class SourceGenerator {
                     let buildFile = PBXBuildFile(reference: referenceGenerator.generate(PBXBuildFile.self, fileReference),
                                                  fileRef: fileReference,
                                                  settings: nil)
-                    allSourceFiles.append(SourceFile(path: filePath, fileReference: fileReference, buildFile: buildFile))
+                    allSourceFiles.append(SourceFile(path: filePath, fileReference: fileReference, buildFile: buildFile, buildPhase: .resources))
                     groupChildren.append(fileReference)
                 }
             }
         }
 
-        let isOutOfBasePath = !path.string.contains(spec.basePath.string)
-        let isRootPath = isOutOfBasePath || path.parent() == spec.basePath
-        let isRootGroup = (isRootSource && !spec.options.createIntermediateGroups) || isRootPath
-        let group = getGroup(path: path, name: isRootSource && !isFile ? targetSource.name : nil, mergingChildren: groupChildren, isRootGroup: isRootGroup)
+        let group = getGroup(path: path, mergingChildren: groupChildren, createIntermediateGroups: spec.options.createIntermediateGroups, isBaseGroup: isBaseGroup)
         if spec.options.createIntermediateGroups {
-            createIntermediaGroups(for: group, at: path)
+            createIntermediaGroups(for: group.reference, at: path)
         }
 
         groups.insert(group, at: 0)
         return (allSourceFiles, groups)
     }
 
+    private func getSourceFiles(targetSource: TargetSource, path: Path) throws -> [SourceFile] {
+
+        let type = targetSource.type ?? (path.isFile || path.extension != nil ? .file : .group)
```

Comment:
This is awesome! This will make our generation way faster since we waited on I/O for the `path.isFile` check for every file in our project.